File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        m = len(flowerbed)
        if n == 0:
            return True
        if m == 1 and flowerbed[0] ==1:
            return False
        for i in range(0,m):
            if flowerbed[i] == 0 and(i == 0 or flowerbed[i-1] == 0)and(i == m-1 or flowerbed[i+1] == 0):
                flowerbed[i] = 1
                n-=1
        return n<=0

File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int result = 0;
        vector<int> localvec = flowerbed;
        int l = flowerbed.size();
        if (l == 1) {
            if (localvec[0] == 0)
                return n <= 1;
            return n == 0;
        }
        if (localvec[0] == 0 && localvec[1] == 0) {
            result++;
            localvec[0] = 1;
        }cout <<localvec[0];
        if (localvec[l - 1] == 0 && localvec[l - 2] == 0) {
            result++;
            localvec[l - 1] = 1;
        }
        for (int i = 1; i < l - 1; i++) {
            if (localvec[i - 1] == 0 && localvec[i] == 0 &&
                localvec[i + 1] == 0) {
                result++;
                localvec[i] = 1;
            }
            cout <<localvec[i];
            //if (result == n)
             //   return true;
        }cout <<localvec[l - 1];
        return result >= n;
    }
};